application/src/main/webapp/vue-app/components/ExoChatApp.vue

```diff
@@ -168,10 +168,12 @@ export default {
       if (selectedContact && selectedContact.fullName && (selectedContact.room || selectedContact.user)) {
         eXo.chat.selectedContact = selectedContact;
         const indexOfRoom = this.contactList.findIndex(contact => contact.room === selectedContact.room || contact.user === selectedContact.user);
+        const index = this.contactList.indexOf(selectedContact);
         if(indexOfRoom < 0) {
           this.contactList.unshift(selectedContact);
         } else {
-          this.contactList.splice(indexOfRoom, 1, selectedContact);
+          this.contactList.splice(indexOfRoom, 0, selectedContact);
```

Comment:
Array.splice will remove an element and add selectedContact, why is it done twice ?

Reply:
actually , this is not the change i made , i will make an update in the 175 line .

Reply:
Array.splice will add the selected contact in the head of the list in the first place , and then remove it from its current index so that it is not duplicated. so in the 175 line Arraysplice just pushed the contact , it didin't remove anything

Reply:
Yes both could be done in one instruction as detailed in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice

Reply:
with array.splice , it's not possible to remove an element from an index and insert another element in another index